### PR TITLE
[DA-178] Add support for whitelist cli changes

### PR DESCRIPTION
**Pull request by thescouser89**
Changes:
- We now have to specify the product for the whitelist
- As such, the CLI had to be changed to accomodate this change

- Instead of implementing the functionality in Bash, the author of the
  commit decided to write part of it in Python since it's easier to
  parse JSON in Python

This PR should be merged after PR #210

### CODE REVIEW

**Comment by thescouser89** [comment]
CLI updated with new endpoint for deleting a particular artifact for a particular product.

File: cli/da-cli-admin.sh [hunk 1]
```diff
@@ -10,14 +10,44 @@ printUsage() {
...
-    echo "$0 list (b[lack]|w[hite])"
-    echo "    List all artifacts in black or white list."
+    echo "$0 list black"
+    echo "    List all artifacts in blacklist."
     echo ""
-    echo "$0 add (b[lack]|w[hite]) GROUP_ID:ARTIFACT_ID:VERSION"
-    echo "    Add artifact GROUP_ID:ARTIFACT_ID:VERSION to black or white list"
+    echo "$0 list white [GROUP_ID:ARTIFACT_ID:VERSION]"
+    echo "    List all artifacts in whitelist."
+    echo "    You can optionally specify which GAV to list"
```
Comment (jbrazdil) [inline cli/da-cli-admin.sh]: I don't get it, what it does?

File: cli/da-cli-admin.sh [hunk 2]
```diff
@@ -10,14 +10,44 @@ printUsage() {
...
     echo ""
-    echo "$0 delete (b[lack]|w[hite]) GROUP_ID:ARTIFACT_ID:VERSION"
-    echo "    Delete artifact GROUP_ID:ARTIFACT_ID:VERSION from black or white list"
+    echo "$0 list whitelist-products [GROUP_ID:ARTIFACT_ID:VERSION]"
+    echo "    List all products in the whitelist"
+    echo "    You can optionally list products which have a particular GAV"
```
Comment (jbrazdil) [inline cli/da-cli-admin.sh]: What does it mean "product which have a particular GAV"?

File: cli/da-cli-admin.sh [hunk 3]
```diff
@@ -10,14 +10,44 @@ printUsage() {
...
+    echo ""
+    echo "$0 list whitelist-ga GROUP_ID:ARTIFACT_ID STATUS"
+    echo "    List all artifacts in the whitelist with a particular GA and status"
```
Comment (jbrazdil) [inline cli/da-cli-admin.sh]: What is the possible status?

File: cli/da-cli-admin.sh [hunk 4]
```diff
@@ -10,14 +10,44 @@ printUsage() {
...
+    echo ""
+    echo "$0 list whitelist-gavs STATUS"
+    echo "    List all artifacts in the whitelist with a particularstatus"
```
Comment (jbrazdil) [inline cli/da-cli-admin.sh]: Typo, missing space ``particularstatus``

File: cli/da-cli-script.py [hunk 5]
```diff
@@ -0,0 +1,349 @@
...
+            product_support = item['supportStatus']
+
+            if show_artifacts:
+                gid = item['gav']['groupId']
+                aid = item['gav']['artifactId']
+                ver = item['gav']['version']
+                print("{}:{}:{} \t {}:{}:{}".format(gid, aid, ver, product,
+                                                    product_version, product_support))
+            else:
+                print("{}:{}:{}".format(product, product_version, product_support))
```
Comment (jbrazdil) [inline cli/da-cli-script.py]: Should the support be separated by ``:`` from ``PRODUCT:VERSION``? Wouldn't be space/tab better?

**Comment by jbrazdil** [comment]
There is some places in the help I think could be explained better.

File: cli/da-cli-admin.sh (hunk 1 above)
Comment (thescouser89) [inline cli/da-cli-admin.sh]: Each artifact for whitelist has to be associated with a product. Will Update the documentation.

**Comment by jbrazdil** [comment]
:+1:

**Pull request merged**